pkg/resources/kafka/kafka.go

```diff
@@ -565,10 +568,45 @@ func (r *Reconciler) reconcileKafkaPod(log logr.Logger, desiredPod *corev1.Pod,
 	return nil
 }
 
-func (r *Reconciler) updateStatusWithDockerImageAndVersion(brokerId int32, brokerConfig *v1beta1.BrokerConfig, log logr.Logger) error {
+func (r *Reconciler) updateStatusWithDockerImageAndVersion(brokerId int32, brokerConfig *v1beta1.BrokerConfig,
+	log logr.Logger) error {
+	const versionRegexGroup = "version"
+	var requestURL string
+	if r.KafkaCluster.Spec.HeadlessServiceEnabled {
+		requestURL =
+			fmt.Sprintf("http://%s-%d."+kafka.HeadlessServiceTemplate+":%d", r.KafkaCluster.GetName(), brokerId, r.KafkaCluster.GetName(), 9020)
+	} else {
+		requestURL = fmt.Sprintf("http://%s-%d:%d", r.KafkaCluster.GetName(), brokerId, 9020)
+	}
+	rsp, err := http.Get(requestURL)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("error during talking to broker-%d", brokerId))
+		return errorfactory.New(errorfactory.BrokersNotReady{}, err, "unable to talk to ...")
+	}
+	defer func() {
+		closeErr := rsp.Body.Close()
+		if closeErr != nil {
+			log.Error(closeErr, "could not close client")
+		}
+	}()
+
+	body, err := ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return err
+	}
+	jmxMetricRegex := regexp.MustCompile(
```

Comment:
This could be made a package level private variable so that the regex is not re-compiled whenever `updateStatusWithDockerImageAndVersion` is invoked.